### ISSUE TITLE
Plug different application protocol

### ISSUE THREAD

**Issue by slinkydeveloper** [tsenart/vegeta#443]
#### Proposal

Allow to use any application protocol more than only HTTP while using vegeta as library.

My idea is to have a `Transport` trait, that provides a `Send()` method. When the `Transport` implementation is instantiated, it gets a specific `Targeter` to create the requests and a specific `ErrorChecker` to check if the response must be treated as a failed response (as in http we treat `res.Code < 200 || res.Code >= 400` as error)

To avoid breaking changes, we can create an `HttpTransport` as default one and various configuration functions (`Connections`, `KeepAlive`, etc) changes those values inside `HttpTransport.client`

Does it sounds good to you? If you're interested, I could provide a PR for it

#### Background

We are using in [knative eventing benchmarks](https://github.com/knative/eventing/tree/master/test/test_images/performance) vegeta to perform load testing.
We would like to [test](https://github.com/knative/eventing-contrib/issues/607) a component that requires to send events through kafka and at same time  we still want to continue to use vegeta as load generator, since it's proven to generate an affordable load

#### Workarounds

Can't do it atm with vegeta.

**Comment by slinkydeveloper** [comment]
Partially related to https://github.com/tsenart/vegeta/issues/431

**Comment by slinkydeveloper** [comment]
This is a wip I'm working on where I've basically rebuilt the vegeta load generation loop, but workers have pluggable send logic https://github.com/slinkydeveloper/loadastic

**Comment by tsenart** [comment]
Thanks for these thoughts, but Vegeta is an HTTP load testing tool and library — making it generic would increase complexity in a way that it would make using it for HTTP harder than I want it it to be.

In cases like this, where I found myself wanting to load test other protocols, I ended up writing a thin Go HTTP server that serves as a [façade](https://en.wikipedia.org/wiki/Facade_pattern) to said protocol. I then direct vegeta at that server.

This may not work for you, but it seems like your project is already covering your needs so I'm glad for that :-)
